### PR TITLE
Toyota: match openpilot standstill check

### PR DESCRIPTION
**Pull request by sshane**

### CODE REVIEW

**Comment by sshane** [comment]
@adeebshihadeh should be safe, lowest wheel speeds ever reported were 0.01 kph which is 1 here.

File: board/safety/safety_toyota.h [hunk 1]
```diff
@@ -106,7 +104,7 @@ static int toyota_rx_hook(CANPacket_t *to_push) {
         int wheel_speed = (GET_BYTE(to_push, i) << 8U) + GET_BYTE(to_push, (i+1U));
         speed += wheel_speed - 0x1a6f;
       }
-      vehicle_moving = ABS(speed / 4) > TOYOTA_STANDSTILL_THRSLD;
+      vehicle_moving = ABS(speed) > 0;
```
Comment (adeebshihadeh) [inline board/safety/safety_toyota.h:107]: no need to add them up anymore, just check this in the for loop

File: board/safety/safety_toyota.h [hunk 2]
```diff
@@ -98,15 +96,13 @@ static int toyota_rx_hook(CANPacket_t *to_push) {
...
 
-    // sample speed
     if (addr == 0xaa) {
-      int speed = 0;
-      // sum 4 wheel speeds
-      for (uint8_t i=0U; i<8U; i+=2U) {
-        int wheel_speed = (GET_BYTE(to_push, i) << 8U) + GET_BYTE(to_push, (i+1U));
-        speed += wheel_speed - 0x1a6f;
+      // check any wheel speed for non-zero value
+      vehicle_moving = false;
```
Comment (adeebshihadeh) [inline board/safety/safety_toyota.h:101]: make sure this is done atomically

File: board/safety/safety_toyota.h [hunk 3]
```diff
@@ -98,15 +96,13 @@ static int toyota_rx_hook(CANPacket_t *to_push) {
...
+      for (uint8_t i = 0U; i < 8U; i += 2U) {
+        int wheel_speed = (GET_BYTE(to_push, i) << 8U) + GET_BYTE(to_push, (i + 1U));
+        vehicle_moving |= wheel_speed != 0x1a6f;
```
Comment (adeebshihadeh) [inline board/safety/safety_toyota.h:104]: use a logical or

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: board/safety/safety_toyota.h (hunk 2 above)
Comment (sshane) [reply board/safety/safety_toyota.h]: Tried to just use `GET_BYTES_04`, but it weirdly returned `0x6f1a6f00` for 0 speed, not `0x6f1a6f1a`.

**Pull request merged**